pkg/resourcemanager/webhook/nodeagentauthorizer/authorizer.go

```diff
@@ -143,20 +145,31 @@ func (a *authorizer) authorizeLease(ctx context.Context, log logr.Logger, machin
 		return auth.DecisionDeny, reason, nil
 	}
 
-	machine := &machinev1alpha1.Machine{}
-	if err := a.sourceClient.Get(ctx, client.ObjectKey{Name: machineName, Namespace: a.machineNamespace}, machine); err != nil {
-		return auth.DecisionDeny, "", fmt.Errorf("error getting machine %q: %w", machineName, err)
+	var nodeName string
+	if a.machineNamespace != nil {
+		machine := &machinev1alpha1.Machine{}
+		if err := a.sourceClient.Get(ctx, client.ObjectKey{Name: machineName, Namespace: *a.machineNamespace}, machine); err != nil {
+			return auth.DecisionDeny, "", fmt.Errorf("error getting machine %q: %w", machineName, err)
+		}
+
+		nodeName = machine.Labels[machinev1alpha1.NodeLabelKey]
+	} else {
+		node := &corev1.Node{}
+		if err := a.targetClient.Get(ctx, client.ObjectKey{Name: machineName}, node); client.IgnoreNotFound(err) != nil {
+			return auth.DecisionDeny, "", fmt.Errorf("error getting node %q: %w", machineName, err)
+		}
+
+		nodeName = node.Name
 	}
 
-	node := machine.Labels[machinev1alpha1.NodeLabelKey]
-	if node == "" {
-		log.Info(`Denying request because the machine does not have a "node" label`, "machineName", machineName)
-		return auth.DecisionDeny, fmt.Sprintf(`expecting "node" label on machine %q`, machineName), nil
+	if nodeName == "" {
+		log.Info("Denying request because no node for the user was found", "machineName", machineName)
```

Comment:
Nit: I find the reference to `user` confusing.

```suggestion
		log.Info(fmt.Sprintf("Denying request because no node related to %q was found", machineName), "machineName", machineName)
```